### PR TITLE
Add file type support and update core to 7.0.2 SDK-4096 

### PR DESCRIPTION
**Pull request by kushCT**
- [x] Add `CleverTap.defineFileVariable("fileVariable")`
- [x] Add `CleverTap.onOneTimeVariablesChanged(handler)`
- [x] Add `CleverTap.onVariablesChangedAndNoDownloadsPending(handler)`
- [x] Add `CleverTap.onceVariablesChangedAndNoDownloadsPending(handler)`
- [x] Add `CleverTap.onFileValueChanged('fileVariable', handler)`
- [x] Support core SDK `7.0.2`
- [x] Update [Variables.md](https://github.com/CleverTap/clevertap-react-native/blob/master/docs/Variables.md)
- [x] Update [ChangeLog.md](https://github.com/CleverTap/clevertap-react-native/blob/master/CHANGELOG.md)
- [x] Update Example App
- [x] Bump lib version to `3.1.0`

### CODE REVIEW

File: Example/android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -63,9 +63,13 @@
                 <action android:name="com.clevertap.PUSH_EVENT" />
             </intent-filter>
         </service>
-        <meta-data android:name="CLEVERTAP_ACCOUNT_ID" android:value="CLEVERTAP_ACCOUNT_ID" />
-        <meta-data android:name="CLEVERTAP_TOKEN" android:value="CLEVERTAP_ACCOUNT_TOKEN" />
+        <meta-data android:name="CLEVERTAP_ACCOUNT_ID" android:value="4WK-RWR-796Z" />
```
Comment (Anush-Shand) [inline Example/android/app/src/main/AndroidManifest.xml:66]: Can these details be removed?

File: Example/ios/Example/Info.plist [hunk 2]
```diff
@@ -23,9 +23,11 @@
 	<key>CFBundleVersion</key>
 	<string>1</string>
 	<key>CleverTapAccountID</key>
-	<string>RKW-R77-KW6Z</string>
+	<string>67Z-RRK-696Z</string>
 	<key>CleverTapToken</key>
-	<string>aa2-032</string>
+	<string>322-1a6</string>
```
Comment (CTLalit) [inline Example/ios/Example/Info.plist:28]: check this.

File: android/settings.gradle [hunk 3]
```diff
@@ -0,0 +1,5 @@
+includeBuild(new File("../../clevertap-android-sdk")) {
+        dependencySubstitution {
+            substitute(module("com.clevertap.android:clevertap-android-sdk")).using(project(":clevertap-core"))
```
Comment (CTLalit) [inline android/settings.gradle:3]: do we take the local android sources for core sdk?

File: Example/ios/Example/Info.plist (hunk 2 above)
Comment (kushCT) [reply Example/ios/Example/Info.plist]: Resolved

File: Example/android/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (kushCT) [reply Example/android/app/src/main/AndroidManifest.xml]: Resolved

File: android/settings.gradle (hunk 3 above)
Comment (vasct) [reply android/settings.gradle:3]: This whole file should be deleted. I has probably been left behind from the custom templates branch.

File: android/src/main/java/com/clevertap/react/CleverTapUtils.java [hunk 4]
```diff
@@ -1,11 +1,15 @@
 package com.clevertap.react;
 
+import android.os.Handler;
```
Comment (vasct) [inline android/src/main/java/com/clevertap/react/CleverTapUtils.java:3]: All the imports added in this file are redundant

File: src/index.js [hunk 5]
```diff
@@ -370,7 +384,7 @@ var CleverTap = {
     * Set the user's location as a latitude,longitude coordinate
     * @param {float} latitude
     * @param {float} longitude
-    */
+    // */
```
Comment (vasct) [inline src/index.js:387]: Redundant comment symbol

File: Example/package-lock.json [hunk 6]
```diff
@@ -1,13598 +0,0 @@
-{
```
Comment (vasct) [inline Example/package-lock.json:1]: Shouldn't we keep this file in the repo ?

File: Example/app/App.js [hunk 7]
```diff
@@ -743,6 +767,11 @@ export default class App extends Component {
           console.log(`variable value for key \'reactnative_var_string\': ${variable}`);
         });
         break;
+        case Actions.GET_FILE_VARIABLE:
```
Comment (nzagorchev) [inline Example/app/App.js:770]: Formatting is a bit off

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,6 +1,27 @@
...
+  * Adds support for custom handshake domains.
+
+* **[Android and iOS Platform]**
+  * Adds support for File Type Variables in Remote Config. Please refer to the [Remote Config Variables](https://github.com/CleverTap/clevertap-ios-sdk/blob/master/docs/Variables.md) doc to read more on how to integrate this in your app.
+  * Adds support for Custom Code Templates. Please refer to the [CustomCodeTemplates.md](https://github.com/CleverTap/clevertap-android-sdk/blob/master/docs/CustomCodeTemplates.md) doc to read more on how to integrate this in your app.
+  * Adds support for custom code in-app templates definitions through a json scheme.
+
+**Bug Fixes**
+* **[Android and iOS Platform]**
+  * Fixed a missing import statement in the index.js file
```
Comment (Anush-Shand) [inline CHANGELOG.md:23]: Let's also mention these github issues in the changelog

https://github.com/CleverTap/clevertap-react-native/issues/431
https://github.com/CleverTap/clevertap-react-native/issues/426

**Review by Anush-Shand — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 5 above)
Comment (piyush-kukadiya) [reply src/index.js]: Fixed

File: android/src/main/java/com/clevertap/react/CleverTapUtils.java (hunk 4 above)
Comment (piyush-kukadiya) [reply android/src/main/java/com/clevertap/react/CleverTapUtils.java]: fixed

File: android/settings.gradle (hunk 3 above)
Comment (piyush-kukadiya) [reply android/settings.gradle]: Fixed

File: Example/app/App.js (hunk 7 above)
Comment (piyush-kukadiya) [reply Example/app/App.js]: Fixed

File: Example/app/App.js [hunk 9]
```diff
@@ -756,6 +785,10 @@ export default class App extends Component {
         console.log(`Creating variables: ${JSON.stringify(variables)}`);
         CleverTap.defineVariables(variables);
         break;
+      case Actions.DEFINE_FILE_VARIABLES:
+          console.log(`Creating file variables: folder1.fileVariable`);
```
[the remark this answers is not in the document]
Comment (piyush-kukadiya) [reply Example/app/App.js]: Fixed

File: Example/app/App.js [hunk 10]
```diff
@@ -766,11 +799,34 @@ export default class App extends Component {
...
+          console.log('onVariablesChangedAndNoDownloadsPending', variables);
+        });
+        break;
+      case Actions.FILES_VARIABLES_CHANGED_AND_DOWNLOADED_ONCE:
+        CleverTap.onceVariablesChangedAndNoDownloadsPending((variables) => {
+          console.log('onceVariablesChangedAndNoDownloadsPending', variables);
+        });
         break;
+      case Actions.FILE_CHANGED:
+      CleverTap.onFileValueChanged('folder1.fileVariable', (variable) => {
```
[the remark this answers is not in the document]
Comment (piyush-kukadiya) [reply Example/app/App.js]: Fixed

File: CHANGELOG.md (hunk 8 above)
Comment (piyush-kukadiya) [reply CHANGELOG.md]: Fixed

File: Example/package-lock.json
[the remark this answers is not in the document]
Comment (kushCT) [reply Example/package-lock.json:1]: Resolved

File: Example/ios/Example.xcodeproj/project.pbxproj [hunk 11]
```diff
@@ -465,6 +472,7 @@
 				81AB9BB82411601600AC10FF /* LaunchScreen.storyboard in Resources */,
 				13B07FBF1A68108700A75B9A /* Images.xcassets in Resources */,
 				685D0A42F74FC38D84284799 /* PrivacyInfo.xcprivacy in Resources */,
+				073713724F3E41CA93910592 /* templates.json in Resources */,
```
[the remark this answers is not in the document]
Comment (kushCT) [reply Example/ios/Example.xcodeproj/project.pbxproj:475]: Resolved

File: Example/ios/Example.xcodeproj/project.pbxproj [hunk 12]
```diff
@@ -285,6 +292,7 @@
 				83CBBA001A601CBA00E9B192 /* Products */,
 				2D16E6871FA4F8E400B85C8A /* Frameworks */,
 				7168044AA71FBFBA4A3B8E24 /* Pods */,
+				6DD9B5B337054DCBBAD650A9 /* Resources */,
```
[the remark this answers is not in the document]
Comment (kushCT) [reply Example/ios/Example.xcodeproj/project.pbxproj:295]: Resolved

File: Example/ios/Example.xcodeproj/project.pbxproj [hunk 13]
```diff
@@ -247,21 +247,28 @@
 			path = NotificationContentSwift;
 			sourceTree = "<group>";
 		};
+		6DD9B5B337054DCBBAD650A9 /* Resources */ = {
```
[the remark this answers is not in the document]
Comment (kushCT) [reply Example/ios/Example.xcodeproj/project.pbxproj:250]: Resolved

File: Example/ios/Example.xcodeproj/project.pbxproj [hunk 14]
```diff
@@ -7,11 +7,10 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
-		12C81FF128A273D97535D021 /* libPods-NotificationServiceSwift.a in Frameworks */ = {isa = PBXBuildFile; fileRef = D06E088C940DCB0A8669EA1D /* libPods-NotificationServiceSwift.a */; };
+		073713724F3E41CA93910592 /* templates.json in Resources */ = {isa = PBXBuildFile; fileRef = E1BA1A91DB434DB49CBCE26A /* templates.json */; };
```
[the remark this answers is not in the document]
Comment (kushCT) [reply Example/ios/Example.xcodeproj/project.pbxproj:10]: Resolved

File: Example/ios/link-assets-manifest.json
[the remark this answers is not in the document]
Comment (kushCT) [reply Example/ios/link-assets-manifest.json:1]: Resolved

**Pull request closed without merging**